src/Configuration/Dotenv/ParsedVariable.hs

```diff
@@ -0,0 +1,55 @@
+module Configuration.Dotenv.ParsedVariable (ParsedVariable(..),
+                                            VName,
+                                            VValue(..),
+                                            VContents,
+                                            VFragment(..),
+                                            interpolateParsedVariables) where
+
+import Control.Monad (foldM)
+#if MIN_VERSION_optparse_applicative(0,13,0)
+import Data.Functor ((<$>))
+#endif
+import Control.Applicative ((<|>))
+import System.Environment (lookupEnv)
+
+data ParsedVariable
+  = ParsedVariable VName VValue deriving (Show, Eq)
+
+type VName = String
+
+data VValue
+  = Unquoted VContents
+  | SingleQuoted VContents
+  | DoubleQuoted VContents deriving (Show, Eq)
+
+type VContents = [VFragment]
+
+data VFragment
+  = VInterpolation String
+  | VLiteral String deriving (Show, Eq)
+
+interpolateParsedVariables :: [ParsedVariable] -> IO [(String, String)]
```

Comment:
I typed up a relatively lengthy resposne to this, did you get it? It seems to have disappeared.